test_density/setup_emap.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# encoding: utf-8
+import numpy as np
+import meld
+from meld.comm import MPICommunicator
+from meld import comm, vault
+import mdtraj as md
+from meld import system
+from meld.remd import ladder, adaptor, leader
+from meld.system.scalers import LinearRamp,ConstantRamp
+import scipy.ndimage
+from simtk import unit as u
+import copy
+
+N_REPLICAS = 8
+N_STEPS = 10000
+BLOCK_SIZE = 10
+def gen_state_templates(index, templates):                                           
+
+    n_templates = len(templates)
+    print((index,n_templates,index%n_templates))
+    a = system.subsystem.SubSystemFromPdbFile(templates[index%n_templates])
+    #Note that it does not matter which forcefield we use here to build
+    #as that information is not passed on, it is used for all the same as
+    #in the setup part of the script
+    b = system.builder.SystemBuilder(forcefield="ff14sbside")
+    c = b.build_system([a])
+    pos = c._coordinates
+    c._box_vectors=np.array([0.,0.,0.])
+    vel = np.zeros_like(pos)
+    alpha = index / (N_REPLICAS - 1.0)
+    energy = 0
+    return system.state.SystemState(pos, vel, alpha, energy,c._box_vectors)
+
+def gen_state(s, index):
+    state = s.get_state_template()
+    state.alpha = index / (N_REPLICAS - 1.0)
+    return state
+def map_potential(emap, threshold, scale_factor):
+    emap_cp = copy.deepcopy(emap)
+    emap = scale_factor * ((emap - threshold) / (emap.max() - threshold))
+    emap_where = np.where(emap <= 0)
+    emap_cp = scale_factor * (1 - (emap_cp - threshold) / (emap_cp.max() - threshold))
+    emap_cp[emap_where[0], emap_where[1], emap_where[2]] = scale_factor
+    return emap_cp
+def setup_system():
+    template = "1ake_centered.pdb"
+    p = meld.SubSystemFromPdbFile(template)
+    b = meld.SystemBuilder(forcefield="ff14sbside")
+    s = b.build_system([p])
+    s.temperature_scaler = meld.ConstantTemperatureScaler(300.0 * u.kelvin)  #
+    n_res = s.residue_numbers[-1]
+    blur_scaler = s.restraints.create_scaler(
+        "linear_blur", alpha_min=0, alpha_max=1, min_blur=0.0, max_blur=0.0
+    )
```

Comment:
This doesn't seem to be doing any blurring?

Reply:
Yes, this is a test system that doesn't require blur to fit, I was trying to compare the result with previous code under the same set up.

Reply:
I see.